### PR TITLE
CHE-4014; remove deprecated RemotePreferenceDao

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?
Removes deprecated RemotePreferenceDao.

### What issues does this PR fix or reference?
#4014 

#### Changelog
Removed deprecated RemotePreferenceDao.

#### Release Notes
N/A

#### Docs PR
N/A

### CODE REVIEW

File: wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/LocalGitUserResolver.java [hunk 1]
```diff
@@ -34,23 +40,29 @@
...
 
-    private final PreferenceDao preferenceDao;
+    private final String                 preferencessUrl;
+    private final HttpJsonRequestFactory requestFactory;
 
     @Inject
-    public LocalGitUserResolver(PreferenceDao preferenceDao) {
-        this.preferenceDao = preferenceDao;
+    public LocalGitUserResolver(@Named("che.api") String apiUrl, HttpJsonRequestFactory requestFactory) {
+        this.preferencessUrl = apiUrl + "/preferences";
```
Comment (skabashnyuk) [inline wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/LocalGitUserResolver.java]: Are you going to reuse this ```preferencessUrl``` parameter?

**Review by skabashnyuk — APPROVED** [review]

File: wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/LocalGitUserResolver.java [hunk 2]
```diff
@@ -34,23 +40,29 @@
...
-                                                                           "git.committer.\\w+");
+            Map<String, String> preferences = requestFactory.fromUrl(preferencessUrl)
+                                                            .useGetMethod()
+                                                            .addQueryParam("filter", "git.committer.\\w+")
+                                                            .request()
+                                                            .asProperties();
             name = preferences.get("git.committer.name");
             email = preferences.get("git.committer.email");
-        } catch (ServerException e) {
+        } catch (ServerException | IOException |
```
Comment (sleshchenko) [inline wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/LocalGitUserResolver.java]: Would it better to catch `ApiException | IOException`?

File: wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/LocalGitUserResolver.java (hunk 1 above)
Comment (sleshchenko) [inline wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/LocalGitUserResolver.java]: Think about adding tests that will ensure that this path really is PreferencesService path
Like this https://github.com/eclipse/che/pull/4386/files#diff-5bb5d6c3bf632ea1c9cce3ed96523af7R71

Comment (sleshchenko) [reply wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/LocalGitUserResolver.java]: Are you going to rename `preferencessUrl` to `preferencesUrl`?

Comment (mshaposhnik) [reply wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/LocalGitUserResolver.java]: Added

File: wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/LocalGitUserResolver.java (hunk 2 above)
Comment (mshaposhnik) [reply wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/LocalGitUserResolver.java]: Yup

**Pull request merged**